### DISCUSSION TITLE
How do I run Tor browser inside firejail ?

### DISCUSSION THREAD

**Discussion by firejailaddssecuirty** [netblue30/firejail#4691] [Q&A] [answered]
I have downloaded the Tor Browser Bundle from their official website & extracted it. The extracted folder is in /home/username/Downloads. Which is the ideal location to run Tor browser from ? How to I run it inside firejail ? Coz as you know I have not installed Tor from the repos so sudo firecfg won't detect it.

**Comment by rusty-snake** [comment, marked as answer]
Asked multiple times. See #4380 for example.

**Reply by firejailaddssecuirty** [reply]
I read that thread carefully. I will try my best to describe my issue. I have copied the Tor browser folder to /home/username/Downloads. Everytime I want to launch the Tor browser I need to do the following

(Step1)
`cd Downloads/tor/`
(Step2)
`firejail ./start-tor-browser.desktop`

The above procedure works but its quite troublesome. I want to create a launcher on my Desktop using which I can launch the Tor browser inside firejail. I know this is not a firejail specific question but if you know how to do this please reply.

**Reply by rusty-snake** [reply]
Add this to your local:
```
private-cwd ${HOME}/Downloads/tor
```

**Reply by firejailaddssecuirty** [reply]
These are the firejail profiles for Tor browser

```
$ ls tor*
tor-browser-ar.profile  tor-browser-en-us.profile  tor-browser-hu.profile  torbrowser-launcher.profile  tor-browser-tr.profile
tor-browser_ar.profile  tor-browser_en-US.profile  tor-browser_hu.profile  tor-browser-nb.profile       tor-browser_tr.profile
tor-browser-ca.profile  tor-browser-es-es.profile  tor-browser-id.profile  tor-browser_nb.profile       tor-browser-vi.profile
tor-browser_ca.profile  tor-browser_es-ES.profile  tor-browser_id.profile  tor-browser-nl.profile       tor-browser_vi.profile
tor-browser-cs.profile  tor-browser-es.profile     tor-browser-is.profile  tor-browser_nl.profile       tor-browser-zh-cn.profile
tor-browser_cs.profile  tor-browser_es.profile     tor-browser_is.profile  tor-browser-pl.profile       tor-browser_zh-CN.profile
tor-browser-da.profile  tor-browser-fa.profile     tor-browser-it.profile  tor-browser_pl.profile       tor-browser-zh-tw.profile
tor-browser_da.profile  tor-browser_fa.profile     tor-browser_it.profile  tor-browser.profile          tor-browser_zh-TW.profile
tor-browser-de.profile  tor-browser-fr.profile     tor-browser-ja.profile  tor-browser-pt-br.profile    torcs.profile
tor-browser_de.profile  tor-browser_fr.profile     tor-browser_ja.profile  tor-browser_pt-BR.profile    tor.profile
tor-browser-el.profile  tor-browser-ga-ie.profile  tor-browser-ka.profile  tor-browser-ru.profile
tor-browser_el.profile  tor-browser_ga-IE.profile  tor-browser_ka.profile  tor-browser_ru.profile
tor-browser-en.profile  tor-browser-he.profile     tor-browser-ko.profile  tor-browser-sv-se.profile
tor-browser_en.profile  tor-browser_he.profile     tor-browser_ko.profile  tor-browser_sv-SE.profile
```
I have renamed the extracted folder as "tor" so what filename should I create for the local ? Is it suppose to be **tor.local** ? 
After creating that local config & adding` private-cwd ${HOME}/Downloads/tor` if I click on the default icon called Tor Browser will Tor start inside firejail automatically ?

---

EDIT by @rusty-snake: code-block formatting

**Reply by rusty-snake** [reply]
> These are the firejail profiles for Tor browser

Well if you look into `tor-browser[_-]{locale}.profile` you see that they are only aliases for `torbrowser-launcher.profile`. If you look at tor.profile you see that it is for tor (the daemon). <sub>torcs.profile is a game.</sub>

Firejail selects the profile based on the name of the program you start and prints it to your terminal if you run `firejail ./start-tor-browser.desktop`: `Reading profile /etc/firejail/start-tor-browser.desktop.profile`. If you look into it, it is a redirect profile to torbrowser-launcher.profile.

**TL;DR:** Therefore is `torbrowser-launcher.local` the right file in any case.

> default icon called Tor Browser 

???

Where is it?

Who created it?

> will Tor[Browser] start inside firejail automatically ?

Open `file:///home/YOU` and see if runs inside firejail.

**Reply by firejailaddssecuirty** [reply]
Success ! This is what I did. I created a local config file named `torbrowser-launcher.local` and added the line `private-cwd ${HOME}/Downloads/tor`. Then I created a shell script on my desktop with the following line 
```
#! /bin/sh
firejail /home/home/Downloads/tor/start-tor-browser.desktop


```
I verified with the setup with
```
$ firejail --list
2545:home::firejail /home/home/.local/opt/firefox/firefox -P 
3648:home::firejail /home/home/Downloads/tor/start-tor-browser.desktop 
```
